test/usecases/table/simulate.py

```diff
@@ -0,0 +1,170 @@
+import numpy as np
+from neuron import gui
+from neuron import h
+
+
+def setup_sim():
+    section = h.Section()
+    section.insert("tbl")
+
+    return section
+
+
+def check_table(y_no_table, y_table, rtol1=1e-2, rtol2=1e-8):
+    assert np.allclose(y_no_table, y_table, rtol=rtol1), f"{y_no_table} != {y_table}"
+    # if the table is not used, we should get identical results, but we don't,
+    # hence the assert below
+    assert not np.allclose(
+        y_no_table, y_table, rtol=rtol2
+    ), f"{y_no_table} != {y_table}"
+
+
+def test_function():
+    section = setup_sim()
+
+    x = np.linspace(-3, 5, 500)
+
+    func = section(0.5).tbl.example_function
+
+    h.c1_tbl = 1
+    h.c2_tbl = 2
+
+    h.usetable_tbl = 0
+    y_no_table = np.array([func(i) for i in x])
+
+    h.usetable_tbl = 1
+    y_table = np.array([func(i) for i in x])
+
+    check_table(y_table, y_no_table, rtol1=1e-4)
+
+    # verify that the table just "clips" the values outside of the range
+    assert func(x[0] - 10) == y_table[0]
+    assert func(x[-1] + 10) == y_table[-1]
+
+    # change parameters and verify
+    h.c1_tbl = 3
+    h.c2_tbl = 4
+
+    h.usetable_tbl = 0
+    y_params_no_table = np.array([func(i) for i in x])
+
+    h.usetable_tbl = 1
+    y_params_table = np.array([func(i) for i in x])
+
+    check_table(y_params_table, y_params_no_table, rtol1=1e-4)
+
+
+def test_procedure():
+    section = setup_sim()
+
+    x = np.linspace(-4, 6, 300)
+
+    proc = section(0.5).tbl.example_procedure
+
+    def modified_args(arg):
+        """
+        Pass arg to the procedure and return new values
+        """
```

Comment:
The comment is either pointing out the immediately obvious, or not drawing my attention to the surprise.